docs/articles/documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.md

```diff
@@ -0,0 +1,328 @@
+
+* the [--cs (--client-scripts)](../command-line-interface.md#--cs-pathpath2---client-scripts-pathpath2) command line option
+
+    ```sh
+    testcafe chrome test.js --client-scripts=mockDate.js,assets/react-helpers.js
+    ```
+
+* the [runner.clientScripts](../programming-interface/runner.md#clientscripts) API method
+
+    ```js
+    runner.clientScripts('mockDate.js', 'scripts/react-helpers.js');
+    ```
+
+* the [clientScripts](../configuration-file.md#clientscripts) configuration file property
+
+    ```json
+    {
+        "clientScripts": ["mockDate.js", "scripts/react-helpers.js"]
+    }
+    ```
+
+## Inject Scripts in Specific Tests
+
+Use the [fixture.clientScripts](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages) and [test.clientScripts](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages) methods to inject scripts into pages visited during a particular test or fixture.
+
+```js
+fixture `My fixture`
+    .page `http://example.com`
+    .clientScripts('assets/jquery.js');
+```
+
+```js
+test
+    ('My test', async t => { /* ... */ })
+    .clientScripts({ module: 'async' });
+```
+
+## Provide Scripts to Inject
+
+You can pass the following arguments to specify the scripts to inject:
+
+* [path to a JavaScript file](#inject-a-javascript-file)
+* [module name](#inject-a-module)
+* [script code](#inject-script-code)
+
+You can also [inject scripts into specific pages](#provide-scripts-for-specific-pages).
+
+> Note that the API methods and configuration option support [multiple arguments](#specify-multiple-scripts).
+
+### Inject a JavaScript File
+
+Specify the JavaScript file path to inject the entire content of this file into the tested pages. You can pass a string or an object with the `path` property.
+
+*CLI*
+
+```text
+testcafe <browser> <tests> --cs <path>
+```
+
+*Programming interface*
+
+```text
+runner.clientScripts( path | { path } )
+```
+
+*Configuration file*
+
+```text
+{
+    "clientScripts": "path" | { path }
+}
+```
+
+*Test API*
+
+```text
+fixture.clientScripts( path | { path } )
+test.clientScripts( path | { path } )
+```
+
+Argument  | Type   | Description
+--------- | ------ | ---------------------------------------------------------------------------
+`path`    | String | The path to the JavaScript file whose content should be injected.
+
+**Examples**
+
+```sh
+testcafe chrome my-tests --cs assets/jquery.js
+```
+
+```js
+runner.clientScripts('assets/jquery.js');
+// or
+runner.clientScripts({ path: 'assets/jquery.js' });
+```
+
+```js
+{
+    "clientScripts": "assets/jquery.js",
+    // or
+    "clientScripts": { "path": "assets/jquery.js" }
+}
+```
+
+> You cannot combine the `path`, [module](#inject-a-module) and [content](#inject-script-code) properties.
+
+### Inject a Module
+
+Specify the module name to inject the module's content into the tested pages. Use a string or an object with the `module` property.
```

Comment:
Add more information about supported module types.

You can inject into the pages  the content of all types of JavaScript modules that is supported by browser ([UMD](https://github.com/umdjs/umd), [ES Harmony](http://wiki.ecmascript.org/doku.php?id=harmony:modules) and etc.)
A lot of popular module from npm has UMD format. So, you can easily use them in your tests.